skiko/src/jvmMain/kotlin/org/jetbrains/skiko/SkiaLayer.jvm.kt

```diff
@@ -135,9 +155,15 @@ actual open class SkiaLayer internal constructor(
         get() = backedLayer.windowHandle
 
     actual var fullscreen: Boolean
-        get() = backedLayer.fullscreen
+        get() = if (fullscreenDispatcher.isFullscreenDispatched) true
+            else backedLayer.fullscreen
         set(value) {
-            backedLayer.fullscreen = value
+            val window = SwingUtilities.getRoot(this) as Window
+            if (!window.isVisible) {
```

Comment:
The logic is very complex, as it have many branching paths (`if`'s). I have found a case where the SkiaLayer doesn't behave as expected:
```
layer.fullscreen = true
window.isVisible = true
layer.fullscreen = false
println(layer.fullscreen) // prints true
```

Also `isFullscreen` will be lying to as, if the user exit fullscreen manually, after we set it programmatically. We should subscribe to resize in AWT (on to window fullscreen in objective c code).